app/scripts/metamask-controller.js

```diff
@@ -872,15 +872,21 @@ export default class MetamaskController extends EventEmitter {
       ),
     });
 
-    this.keyringController =
-      this.coreKeyringController.getEthKeyringController();
-
-    this.keyringController.memStore.subscribe((state) =>
-      this._onKeyringControllerUpdate(state),
+    keyringControllerMessenger.subscribe('KeyringController:unlock', () =>
+      this._onUnlock(),
+    );
+    keyringControllerMessenger.subscribe('KeyringController:lock', () =>
+      this._onLock(),
+    );
+    keyringControllerMessenger.subscribe(
+      'KeyringController:stateChange',
+      (state) => {
+        this._onKeyringControllerUpdate(state);
+      },
     );
```

Comment:
The keyring controller messenger is _just_ meant to be used by the keyring controller. Here in the MetaMask controller we can use the top-level messenger:

```suggestion
    this.controllerMessenger.subscribe('KeyringController:unlock', () =>
      this._onUnlock(),
    );
    this.controllerMessenger.subscribe('KeyringController:lock', () =>
      this._onLock(),
    );
    this.controllerMessenger.subscribe(
      'KeyringController:stateChange',
      (state) => {
        this._onKeyringControllerUpdate(state);
      },
    );
```